app.js

```diff
@@ -52,4 +52,337 @@ document.addEventListener('DOMContentLoaded', () => {
+
+  function pop0() {
+    projects(0, projectItems);
+    content.style.display = 'grid';
+    grand.style.display = 'block';
+  }
+
+  function pop1() {
+    projects(1, projectItems);
+    content.style.display = 'grid';
+    grand.style.display = 'block';
+  }
+
+  function pop2() {
+    projects(2, projectItems);
+    content.style.display = 'grid';
+    grand.style.display = 'block';
+  }
+
+  function pop3() {
+    projects(3, projectItems);
+    content.style.display = 'grid';
+    grand.style.display = 'block';
+  }
+
+  function pop4() {
+    projects(4, projectItems);
+    content.style.display = 'grid';
+    grand.style.display = 'block';
+  }
+
+  function pop5() {
+    projects(5, projectItems);
+    content.style.display = 'grid';
+    grand.style.display = 'block';
+  }
+
+  function pop6() {
+    projects(6, projectItems);
+    content.style.display = 'grid';
+    grand.style.display = 'block';
+  }
+
+  document.querySelector('#btn-0').addEventListener('click', pop0);
+
+  document.querySelector('#btn-1').addEventListener('click', pop1);
+
+  document.querySelector('#btn-2').addEventListener('click', pop2);
+
+  document.querySelector('#btn-3').addEventListener('click', pop3);
+
+  document.querySelector('#btn-4').addEventListener('click', pop4);
+
+  document.querySelector('#btn-5').addEventListener('click', pop5);
+
+  document.querySelector('#btn-6').addEventListener('click', pop6);
+
+  // Validation of Email
+  const form = document.querySelector('#form');
+  const email = document.querySelector('#email');
+  const error = document.querySelector('.error-message');
+
+  // Display Error or Sucess Message
+  function message(input) {
+    const lowCase = input.toLowerCase();
+    const regex = /@+./;
+
+    if (lowCase !== input) {
+      error.innerText = 'Email Should be in Lower Case';
+      error.className = 'flash error';
+      return false;
+    }
+
+    if (!regex.test(input)) {
+      error.innerText = 'Email should have @ and .';
+      error.className = 'flash error';
+    }
+
+    if (lowCase === input) {
+      error.innerText = 'Thank you for Submitting';
+      error.className = 'flash sucess';
+      form.submit();
+      return true;
+    }
+    return input;
+  }
+
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    message(email.value);
+  });
+
+  // Local Storage
+  const name = document.querySelector('#name');
+  const msg = document.querySelector('#msg');
+
+  const storedValues = {
+    name: '',
+    mail: '',
+    message: '',
+  };
+
+  function fillStorage() {
+    storedValues.firstName = name.value;
+    storedValues.mail = email.value;
+    storedValues.message = msg.value;
+    localStorage.setItem('data', JSON.stringify(storedValues));
+  }
+
+  // Fill form with Stored values
+  function collectValues() {
+    const collect = JSON.parse(localStorage.getItem('data'));
+    name.value = collect.name;
+    email.value = collect.mail;
+    msg.value = collect.message;
+  }
```

Comment:
- [ ] Your storage is working almost perfect. The email and text message are working fine but your name is not persisting after the page reload. Kindly make sure you are getting the values from the correct attributes.